test/core/networking/server_test.dart

```diff
@@ -0,0 +1,296 @@
+import 'dart:io';
+import 'package:test/test.dart';
+import 'package:http/http.dart' as http;
+
+import 'package:flyde/core/networking/server.dart';
+import 'package:flyde/core/networking/websockets/session.dart';
+
+import '../../helpers/value_hook.dart';
+import '../../helpers/wait_for.dart';
+
+void main() {
+  test('Can receive http request', () async {
+    final VHook<bool?> received = VHook<bool?>(null);
+
+    final WebServer server = WebServer(
+      InternetAddress.loopbackIPv4,
+      0,
+      httpOnRequest: (HttpRequest req) {
+        received.set(true);
+        req.response.statusCode = 404;
+        req.response.close();
+      },
+    );
```

Comment:
Please create a helper function which creates a webserver object and awaits `ready`. You can attach the different handlers later on.

```dart
Future<WebServer> _createMockServer({bool waitForReady = true}) async {
    final server = WebServer(InternetAddress.loopbackIPv4, 0);
    if (waitForReady) await server.ready;
    return server;
}

final server = await _createMockServer();
server.onHttpRequest = () {};
```